packages/context/src/__tests__/unit/context-view.unit.ts

```diff
@@ -0,0 +1,116 @@
+// Copyright IBM Corp. 2018. All Rights Reserved.
+// Node module: @loopback/context
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {expect} from '@loopback/testlab';
+import {Binding, BindingScope, filterByTag, Context, ContextView} from '../..';
+
+describe('ContextView', () => {
+  let app: Context;
+  let server: Context;
+
+  let bindings: Binding<unknown>[];
+  let contextView: ContextView;
+
+  beforeEach(givenContextView);
+
+  it('tracks bindings', () => {
+    expect(contextView.bindings).to.eql(bindings);
+  });
+
+  it('resolves bindings', async () => {
+    expect(await contextView.resolve()).to.eql(['BAR', 'FOO']);
```

Comment:
what is the difference between function `resolve()` and `values()`?

Reply:
`resolve()` is used for `@inject.*` to resolve sync/async values. `values()` honors the cache.